storage/blob_test.go

```diff
@@ -389,6 +390,41 @@ func (s *StorageBlobSuite) TestGetBlobRange(c *chk.C) {
 	}
 }
 
+func (s *StorageBlobSuite) TestCreateBlockBlobFromReader(c *chk.C) {
+	cli := getBlobClient(c)
+	cnt := randContainer()
+	c.Assert(cli.CreateContainer(cnt, ContainerAccessTypePrivate), chk.IsNil)
+	defer cli.deleteContainer(cnt)
+
+	name := randString(20)
+	data := randBytes(8888)
+	c.Assert(cli.CreateBlockBlobFromReader(cnt, name, uint64(len(data)), bytes.NewReader(data)), chk.IsNil)
+
+	body, err := cli.GetBlob(cnt, name)
+	c.Assert(err, chk.IsNil)
+	gotData, err := ioutil.ReadAll(body)
+	body.Close()
+
+	c.Assert(err, chk.IsNil)
+	c.Assert(gotData, chk.DeepEquals, data)
+}
+
+func (s *StorageBlobSuite) TestCreateBlockBlobFromReaderWithShortData(c *chk.C) {
+	cli := getBlobClient(c)
+	cnt := randContainer()
+	c.Assert(cli.CreateContainer(cnt, ContainerAccessTypePrivate), chk.IsNil)
+	defer cli.deleteContainer(cnt)
+
+	name := randString(20)
+	data := randBytes(8888)
+	err := cli.CreateBlockBlobFromReader(cnt, name, 9999, bytes.NewReader(data))
```

Comment:
this test is now pretty much the same thing as the other test, right? Also you're saying I'm sending 9999 bytes but `data` is 8888 bytes.